fix: updated sample full permissive IAM policy for AWS Batch (#1012)
seqeralabs/docs: platform-cloud/docs/compute-envs/aws-batch.md

Restricting the `batch` actions using resource tags requires that you set the appropriate tags on each Seqera pipeline when configuring it in Platform. Forgetting to set the tag will cause the pipeline to fail to run.
:::

> The job definition and job name resources cannot be restricted to specific names, as Seqera creates job definitions and jobs with dynamic names. Therefore, the wildcard `*` must be used in the name of these resources. `batch:SubmitJob` requires permission on both job definitions and job queues, so make sure to include both ARNs in the `Resource` array.

Reviewer: ```suggestion
The job definition and job name resources cannot be restricted to specific names, as Seqera creates job definitions and jobs with dynamic names. Therefore, the wildcard `*` must be used in the name of these resources. In addition, `batch:SubmitJob` requires permission on both job definitions and job queues, so make sure to include both ARNs in the `Resource` array.
```

### S3 access (optional)

Seqera automatically attempts to fetch a list of S3 buckets available in the AWS account connected to Platform, to provide them in a dropdown menu to be used as Nextflow working directory and make the Compute Environment creation smoother: this feature is optional and users can type the bucket name manually when setting up a Compute Environment. To allow Platform to fetch the list of buckets in the account the `s3:ListAllMyBuckets` action can be added: it must have the `Resource` field set to `*`, as shown in the generic policy at the beginning of this document.

Seqera offers several products to manipulate data on AWS S3 buckets, such as [Studios](../studios/overview) and [Data Explorer](../data/data-explorer); if these features are not used the related permissions can be omitted.

> The IAM policy can be scoped down to only allow limited Read/Write permissions in certain S3 buckets used by Studios/Data Explorer. In addition, the policy must include permission to check the region and list the content of the S3 bucket used as Nextflow work directory. We also recommend allowing `s3:GetObject` permission on the work directory path to allow fetching Nextflow log files.

Reviewer: ```suggestion
The IAM policy can be scoped down to only allow limited read/write permissions in certain S3 buckets used by Studios/Data Explorer. In addition, the policy must include permission to check the region and list the content of the S3 bucket used as Nextflow work directory. We also recommend granting the `s3:GetObject` permission on the work directory path to fetch Nextflow log files.
```

:::note

> If you opted to create a separate S3 bucket only for Nextflow work directories, there is no need for the IAM user to have Read/Write access to it: if Platform is allowed to manage resources (using Batch Forge) the IAM roles automatically created will have the necessary permissions; if you set up the compute environment manually, you can create the required IAM roles with the necessary permissions as detailed in the [manual AWS Batch setup documentation](../enterprise/advanced-topics/manual-aws-batch-setup).

Reviewer: ```suggestion
If you opted to create a separate S3 bucket only for Nextflow work directories, there is no need for the IAM user to have read/write access to it. If Seqera is allowed to manage resources (using Batch Forge) the IAM roles automatically created will have the necessary permissions. 

If you set up the compute environment manually, you can create the required IAM roles with the necessary permissions as detailed in the [manual AWS Batch setup documentation](../enterprise/advanced-topics/manual-aws-batch-setup).
```
